src/bottom-bar/approve-button/approve-button.js

```diff
@@ -0,0 +1,62 @@
+import { useAlert } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { Button } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+import { useWorkflowContext } from '../../workflow-context/index.js'
+import { ApproveModal } from './approve-modal/index.js'
+import { useApproveData } from './use-approve-data.js'
+
+const ApproveButton = ({ disabled }) => {
+    const [approveData, { loading, error: approveError }] = useApproveData()
+    const { show: showApprovalSuccess } = useAlert(i18n.t('Approval saved'))
+
+    // state
+    const [unexpectedError, setUnexpectedError] = useState(null)
+    const [showApproveModal, setShowApproveModal] = useState(false)
+    const { params, refresh } = useWorkflowContext()
+
+    // derived state
+    const error = approveError || unexpectedError
+    const showApprovalDialog = () => setShowApproveModal(true)
+    const hideApprovalDialog = () => setShowApproveModal(false)
+
+    return (
+        <>
+            <Button
+                primary
+                small
+                disabled={disabled}
+                onClick={showApprovalDialog}
+            >
+                {i18n.t('Approve')}
+            </Button>
+
+            {showApproveModal && (
+                <ApproveModal
+                    error={error}
+                    loading={loading}
+                    onApprove={async () => {
+                        try {
+                            const { wf, pe, ou } = params
+                            await approveData({ wf, pe, ou })
```

Comment:
Unfortunately because of the way `useDataMutation` currently works, this await will never resolve. Instead, we have to use the `onComplete` and `onError` callbacks provided to that hook